pyquil/tests/test_quil.py

```diff
@@ -1170,3 +1170,15 @@ def test_subtracting_memory_regions():
     assert isinstance(parsed_param, Sub)
     assert parsed_param.op1 == alpha
     assert parsed_param.op2 == beta
+
+
+def test_out_of_bounds_memory():
+    r = Program().declare('ro', 'BIT', 1)
+    with pytest.raises(IndexError):
+        r[1]
+
+
+@pytest.mark.timeout(5)
```

Comment:
what is this about?

Reply:
Previously this kinda thing would loop indefinitely. An infinite loop is not caught by pytest as a failure. With my patch, the test should complete in well under 5s. If it doesn't, something is broken and it is picked up by pytest as a failure (rather than running infinintely).

Reply:
That makes sense, but it's concerning that this simple test could loop indefinitely... right?

Reply:
Well that's what spurred the PR. The PR prevents the infinite loop. The test makes sure future changes don't reintroduce that behaviour.

Reply:
Aight, I buy it. I'll merge.